Only require menuinst 2.4.0 for macOS

mrclary opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->
Fixes: spyder-ide/spyder#25373
Fixes: spyder-ide/spyder#23206
<!--
Please add any other relevant info below:
-->
* Add NS media keys to info.plist for macOS
* Constrain menuinst >=2.4.0 for macOS only
* Improve version comparison (not lexicographical)
* Fixed issue where site.py was not properly patched for installers
* Updated json schema and improved compliance

regro-cf-autotick-bot committed:
updated v5.4.4

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.25.0, conda-smithy 3.24.0, and conda-forge-pinning 2023.07.18.22.54.11

dalthviz committed:
Update dependencies

* `spyder-kernels` to >=2.4.4
* `pylint-venv` to >=3.0.2
* Skip `ipython` versions - !=8.11.0,!=8.12.0,!=8.12.1
* `python-lsp-server` to >=1.7.4

dalthviz committed:
Merge pull request #145 from regro-cf-autotick-bot/5.4.4_h272398

spyder v5.4.4

regro-cf-autotick-bot committed:
updated v5.4.5

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.26.1, conda-smithy 3.24.1, and conda-forge-pinning 2023.08.29.11.33.45

dalthviz committed:
Copy spyder_reset.ico but use for the menu entry spyder.ico for the moment

Co-authored-by: Bradley Kreider <1699478+bkreider@users.noreply.github.com>

ccordoba12 committed:
Merge pull request #147 from regro-cf-autotick-bot/5.4.5_hed8a61

spyder v5.4.5

regro-cf-autotick-bot committed:
updated v5.5.0

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.28.0, and conda-forge-pinning 2023.11.07.18.09.01

dalthviz committed:
Update dependencies 

* ipython != 8.17.1
* pylint <3.1
* pyqt >=5.10
* python-lsp-server >=1.9.0,<1.10.0
* qdarkstyle >=3.2.0,<3.3.0
* qtconsole >=5.5.0,<5.6.0
* spyder-kernels >=2.5.0,<2.6.0

dalthviz committed:
Merge pull request #149 from regro-cf-autotick-bot/5.5.0_hfebfd2

spyder v5.5.0

dalthviz committed:
Bump jedi upper constraint to <0.20.0

dalthviz committed:
Update `ipython` and `pyqtwebengine` constraints 

* ipython >=8.12.2,<9.0.0,!=8.17.1
* pyqtwebengine >=5.10,<5.16

ccordoba12 committed:
Merge pull request #150 from dalthviz/jedi_0.19

PR: Bump `jedi` upper constraint to <0.20.0 and improve constraints for `ipython` and `pyqtwebengine`

ccordoba12 committed:
Add fine-grained selector to applaunchservices

This fixes our arm64 builds on Mac.

Commit:
MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.29.0, and conda-forge-pinning 2023.11.22.22.25.59

ccordoba12 committed:
Fix skip condition

Commit:
MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.29.0, and conda-forge-pinning 2023.11.22.22.25.59

ccordoba12 committed:
Remove condition because it's not necessary due to the skip

ccordoba12 committed:
Merge pull request #151 from ccordoba12/fix-osx-dep

Skip osx-arm64 build for Python 3.8

regro-cf-autotick-bot committed:
Rebuild for python312

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.30.0, and conda-forge-pinning 2023.11.23.15.09.04

ccordoba12 committed:
Merge pull request #152 from regro-cf-autotick-bot/rebuild-python312-0-1_h9e0877

Rebuild for python312

regro-cf-autotick-bot committed:
updated v5.5.1

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.28.4, conda-smithy 3.30.4, and conda-forge-pinning 2024.02.06.16.43.21

ccordoba12 committed:
Update dependencies

github-actions[bot] committed:
spyder v5.5.1 (#159)

automerged PR by conda-forge/automerge-action

regro-cf-autotick-bot committed:
updated v5.5.2

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 24.1.2, conda-smithy 3.31.1, and conda-forge-pinning 2024.03.12.21.52.52

dalthviz committed:
Update dependencies

- qtawesome >=1.2.1,<1.3.0
- spyder-kernels >=2.5.1,<2.6.0

dalthviz committed:
Merge pull request #162 from regro-cf-autotick-bot/5.5.2_h7e9df6

spyder v5.5.2

conda-forge-webservices[bot] committed:
dummy commit for rerendering

Commit:
ENH updated version to 5.5.3

Commit:
MNT: Re-rendered with conda-build 24.1.2, conda-smithy 3.32.0, and conda-forge-pinning 2024.03.16.19.50.45

ccordoba12 committed:
Add run_test.py file to check conda_activate.sh is executable

ccordoba12 committed:
Merge pull request #165 from conda-forge-admin/conda_forge_admin_164

ENH: update package version

ytausch committed:
[ci skip] use bot.inspection = disabled

https://github.com/regro/cf-scripts/issues/2272#issuecomment-2004119968

beckermr committed:
Merge pull request #168 from ytausch/patch-1

[ci skip] use bot.inspection = disabled

regro-cf-autotick-bot committed:
updated v5.5.4

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 24.3.0, conda-smithy 3.34.1, and conda-forge-pinning 2024.04.09.19.22.01

dalthviz committed:
Update dependencies

- pylint >=3.1,<4
- python-lsp-server >=1.11.0,<1.12.0 
- pyzmq >=24.0.0 
- qtawesome >=1.3.1,<1.4.0

dalthviz committed:
Merge pull request #169 from regro-cf-autotick-bot/5.5.4_hf3fc89

spyder v5.5.4

conda-forge-curator[bot] committed:
[ci skip] [skip ci] [cf admin skip] ***NO_CI*** admin migration CondaForgeAutomergeUpdate

regro-cf-autotick-bot committed:
updated v5.5.5

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 24.5.1, conda-smithy 3.36.1, and conda-forge-pinning 2024.06.12.03.20.45

dalthviz committed:
Update dependencies

- spyder-kernels >=2.5.2,<2.6.0

dalthviz committed:
Merge pull request #173 from regro-cf-autotick-bot/5.5.5_h97a05a

spyder v5.5.5

regro-cf-autotick-bot committed:
updated v5.5.6

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.39.0, and conda-forge-pinning 2024.08.26.19.18.07

dalthviz committed:
Update dependencies

- importlib-metadata >=4.6.0
- python-lsp-server >=1.12.0,<1.13.0

dalthviz committed:
Merge pull request #182 from regro-cf-autotick-bot/5.5.6_h8c3564

spyder v5.5.6

dalthviz committed:
Update recipe files to 6.0.0 version

Co-authored-by: Carlos Cordoba <ccordoba12@gmail.com>
Co-authored-by: Ryan Clary <9618975+mrclary@users.noreply.github.com>

Commit:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.39.1, and conda-forge-pinning 2024.09.02.15.16.22

dalthviz committed:
Apply suggestions from code review

Co-authored-by: Carlos Cordoba <ccordoba12@gmail.com>

dalthviz committed:
Remove conda_build_config.yaml file

Commit:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.39.1, and conda-forge-pinning 2024.09.02.15.16.22

dalthviz committed:
Merge pull request #184 from dalthviz/600_update

Update recipe files to 6.0.0 version

mrclary committed:
Check menuinst version and take appropriate action.

mrclary committed:
Update build number

ccordoba12 committed:
Merge pull request #185 from mrclary/menuinst-version

Use v1 json spec for Windows menu entry and do not create shortcut on Posix for `menuinst` less than 2.1.2

conda-forge-webservices[bot] committed:
dummy commit for rerendering

Commit:
ENH updated version to 6.0.1

Commit:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.09.23.09.40.07

dalthviz committed:
Merge pull request #187 from conda-forge-admin/conda_forge_admin_186

ENH: update package version

mrclary committed:
Fix issue spyder-ide/spyder#22566
* Do not check menuinst version for conda-based apps
* Do not create shortcut if there is an error retrieving the menuinst version.

mrclary committed:
Bump build number

github-actions[bot] committed:
Fix issue where retrieving menuinst version resulted in an error (#188)

automerged PR by conda-forge/automerge-action

ccordoba12 committed:
Add rc channel for betas and rc releases

Commit:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.40.1, and conda-forge-pinning 2024.09.24.21.28.27

ccordoba12 committed:
Bump build number and change channel order

Commit:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.40.1, and conda-forge-pinning 2024.09.24.21.28.27

ccordoba12 committed:
Merge pull request #189 from ccordoba12/add-rc-channel

Add `rc` channel for betas and release candidates

ccordoba12 committed:
Update with changes in main

Commit:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.43.0, and conda-forge-pinning 2024.10.29.12.41.35

ccordoba12 committed:
Merge pull request #194 from ccordoba12/update-rc

Update with changes in main

dalthviz committed:
Update to 6.0.2rc1

- spyder-kernels >=3.0.1,<3.1.0
- qtconsole >=5.6.1,<5.7.0

dalthviz committed:
Merge pull request #195 from dalthviz/update_6.0.2rc1

PR: Update to 6.0.2rc1

conda-forge-admin committed:
[ci skip] [skip ci] [cf admin skip] ***NO_CI*** admin migration RemoveAutomergeAndRerender

mrclary committed:
Create spyder-base package that does not depend on pyqt.
Use jinja variable for build number and bump.
Remove reference to python version in tests to ensure that spyder package has same hash regardless of runner.

ccordoba12 committed:
Update rc channel

dalthviz committed:
Update to 6.0.3rc1

Commit:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.44.9, and conda-forge-pinning 2024.12.06.19.20.35

dalthviz committed:
Merge pull request #206 from ccordoba12/update_6.0.3rc1

Update to 6.0.3rc1

dalthviz committed:
Update rc channel

dalthviz committed:
Update to 6.0.3rc2

dalthviz committed:
Merge pull request #209 from dalthviz/update_6.0.3rc2

Update to 6.0.3rc2

dalthviz committed:
Update to 6.0.4rc1

Commit:
MNT: Re-rendered with conda-build 25.1.1, conda-smithy 3.45.4, and conda-forge-pinning 2025.02.05.19.16.07

dalthviz committed:
Merge pull request #213 from mrclary/spyder-base-rc

RC version of PR 203

dalthviz committed:
Update rc channel

dalthviz committed:
Update to 6.0.4rc1

dalthviz committed:
Merge pull request #214 from dalthviz/update_6.0.4rc1

PR: Additional fixes for 6.0.4rc1

dalthviz committed:
Update to 6.0.5rc1

Commit:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.47.0, and conda-forge-pinning 2025.03.25.16.41.26

dalthviz committed:
Update dependencies

- ipython >=8.13.0,<9.0.0,!=8.17.1
- qtawesome >=1.4.0,<1.5.0

dalthviz committed:
Merge pull request #222 from dalthviz/update_6.0.5rc1

Update to 6.0.5rc1

dalthviz committed:
Update to 6.0.6rc1

Commit:
MNT: Re-rendered with conda-build 25.4.2, conda-smithy 3.48.1, and conda-forge-pinning 2025.05.13.09.46.09

dalthviz committed:
Merge pull request #235 from dalthviz/update_6.0.6rc1

Update to 6.0.6rc1

dalthviz committed:
Update to 6.0.7rc1

dalthviz committed:
Merge pull request #238 from dalthviz/update_6.0.7rc1

Update to 6.0.7rc1

dalthviz committed:
Update to 6.0.8rc1

Commit:
MNT: Re-rendered with conda-build 25.7.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.08.25.16.21.11

dalthviz committed:
Merge pull request #251 from dalthviz/update_6.0.8rc1

Update to 6.0.8rc1

dalthviz committed:
Update to 6.1.0b1 (#259)

* Update to 6.1.0b1

* MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.11.23.09.5

* Try setting conda_build_tool

* MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.12.13.06.3

* Try setting conda_solver

* MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.12.18.51.0

* Undo conda_solver change. Tests constraint relaxation

* MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.12.18.55.1

* Testing tighten version constraint

* Test changing channel_sources priority order

* MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.12.18.55.1

* Restore spyder-kernels constraint. Add python-lsp-ruff constraint

---------

Co-authored-by: conda-forge-webservices[bot] <91080706+conda-forge-webservices[bot]@users.noreply.github.com>

dalthviz committed:
Update to 6.1.0b2

dalthviz committed:
Merge pull request #260 from dalthviz/update_6.1.0b2

Update to 6.1.0b2

dalthviz committed:
Update to 6.1.0rc1

dalthviz committed:
Merge pull request #261 from dalthviz/update_6.1.0rc1

Update to 6.1.0rc1

dalthviz committed:
Update to 6.1.0rc2

Commit:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.01.17.11.5

dalthviz committed:
Merge pull request #262 from dalthviz/update_6.1.0rc2

Update to 6.1.0rc2

mrclary committed:
Add NS media keys to info.plist for macOS.
Requires menuinst >= 2.4.0

dalthviz committed:
Update rc channel to 6.1.1rc1

dalthviz committed:
Add NS media keys to info.plist for macOS.

Co-authored-by: Ryan Clary <9618975+mrclary@users.noreply.github.com>

dalthviz committed:
Testing

Commit:
MNT: Re-rendered with conda-smithy 3.53.1 and conda-forge-pinning 2025.11.19.18.25.3

dalthviz committed:
Testing

dalthviz committed:
Merge pull request #265 from dalthviz/update_6.1.1rc1

Update to 6.1.1rc1

mrclary committed:
Only require 2.4.0 for macOS

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of `<output 1` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 1` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 1` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).
 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.
 * ℹ️ top-level output has some malformed specs:
- In section run: `spyder-base ==6.1.0a4=*3`
- In section run_constrained: `spyder ==6.1.0a4=*3`
spyder output has some malformed specs:
- In section run: `spyder-base ==6.1.0a4=*3`
Requirement spec fields should match the syntax `name [version [build]]`to avoid known issues in conda-build. For example, instead of `name =version=build`, use `name version.* build`. There should be no spaces between version operators and versions either: `python >= 3.8` should be `python >=3.8`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/19652061818](https://github.com/conda-forge/conda-forge-webservices/actions/runs/19652061818). Examine the logs at this URL for more detail.</sub>

mrclary committed:
Improve version comparison (not lexicographical)

mrclary commented:
> I think the failures are related with the schema validation manuinst does for the .json related with the menus definitions. When updating the recipe for Spyder 6.1.1 I had similar issue although only for the Windows menus. Had to do some changes in the menu definition to make the schema validation pass (https://github.com/conda-forge/spyder-feedstock/pull/267/files#diff-56fc967aafad42eca60eb84cc121c8c9285c767b74f7734d9a884b79777f0b41) but also could be worthy to see what would be the schema version most suited to be used by use? I think now menuinst defaults to a schema version where our current menu definitions are not supported? 🤔

I'll take a look.

mrclary committed:
Update schema specification.

Before conda/schemas#85
"WARNING: Could not fetch 'https://schemas.conda.org/menuinst/menuinst-1-1-2.schema.json', status code 404"
"RuntimeError: Could not build validator for schema 'https://schemas.conda.org/menuinst/menuinst-1-1-2.schema.json'."

Original error, prior to this commit:
"WARNING: Known wrong value for $schema, defaulting to 'https://schemas.conda.org/menuinst/menuinst-1-1-0.schema.json'"
macOS: "jsonschema.exceptions.ValidationError: Additional properties are not allowed ('NSAudioCaptureUsageDescription', 'NSCameraUsageDescription', 'NSMainCameraUsageDescription', 'NSMicrophoneUsageDescription' were unexpected)"
Windows:
"jsonschema.exceptions.ValidationError: Additional properties are not allowed ('app_user_model_id', 'desktop', 'file_extensions' were unexpected)"

mrclary committed:
site.py was not properly patched due to redundant directory path.
How did this get missed?
spyder-ide/spyder#24773
conda-forge/spyder-feedstock#248

mrclary committed:
Improve version comparison (not lexicographical)

dalthviz committed:
Update to 6.1.2rc1

dalthviz committed:
Merge pull request #273 from dalthviz/update_6.1.2rc1

Update to 6.1.2rc1

dalthviz committed:
Update to 6.1.3rc1

Commit:
MNT: Re-rendered with conda-smithy 3.54.2 and conda-forge-pinning 2026.02.09.23.12.1

dalthviz committed:
Merge pull request #277 from dalthviz/update_6.1.3rc1

Update to 6.1.3rc1

dalthviz committed:
Update to 6.1.3rc2

dalthviz committed:
Merge pull request #278 from dalthviz/update_6.1.3rc2

Update to 6.1.3rc2

ccordoba12 commented:
@mrclary, we can merge this one now, i.e. before waiting to release 6.1.4rc1, can't we?

ccordoba12 commented:
Also, perhaps we should move this PR to the `rc` branch?

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict. Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the `@` notation in a comment) if you believe this is a bug.

mrclary commented:
Closing in favor of #281 in order to merge to `rc` instead of `dev`
